frontend/src/components/Admin/AdminShoutouts/AdminShoutouts.tsx

```diff
@@ -1,41 +1,92 @@
 import React, { useState, useEffect } from 'react';
-import { Item, Card } from 'semantic-ui-react';
+import { Form, Item, Card } from 'semantic-ui-react';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+import { Emitters } from '../../../utils';
 import { ShoutoutsAPI, Shoutout } from '../../../API/ShoutoutsAPI';
 import styles from './AdminShoutouts.module.css';
 
 const AdminShoutouts: React.FC = () => {
   const [shoutouts, setShoutouts] = useState<Shoutout[]>([]);
+  const [displayShoutouts, setDisplayShoutouts] = useState<Shoutout[]>([]);
   useEffect(() => {
-    ShoutoutsAPI.getAllShoutouts().then((shoutouts) => setShoutouts(shoutouts));
+    ShoutoutsAPI.getAllShoutouts().then((shoutouts) => {
+      setShoutouts(shoutouts);
+      setDisplayShoutouts(shoutouts);
+    });
   }, []);
+  const [earlyDate, setEarlyDate] = useState<Date>(new Date('2022/01/01'));
+  const [lastDate, setLastDate] = useState<Date>(new Date());
 
   const fromString = (shoutout: Shoutout): string => {
+    const dateString = ` (Date: ${new Date(shoutout.timestamp).toDateString()})`;
     if (!shoutout.isAnon) {
       const { giver } = shoutout;
-      return `From: ${giver?.firstName} ${giver?.lastName} (${giver.email})`;
+      return `From: ${giver?.firstName} ${giver?.lastName} ${dateString}`;
+    }
+    return `From: Anonymous${dateString}`;
+  };
+
+  const updateShoutouts = () => {
+    if (lastDate < earlyDate) {
+      Emitters.generalError.emit({
+        headerMsg: 'Invalid Date Range',
+        contentMsg: 'Please make sure the latest shoutout date is after the earliest shoutout date.'
+      });
+    } else {
+      setDisplayShoutouts(
+        shoutouts.filter(
+          (shoutout) =>
+            new Date(shoutout.timestamp) >= earlyDate && new Date(shoutout.timestamp) <= lastDate
```

Comment:
Avoiding code duplication/extra instantiating of a Date object, this should probably be best stored in a `const` variable right before the function call and used twice here.